net/tstun/wrap.go

```diff
@@ -721,3 +732,15 @@ func (t *Wrapper) InjectOutbound(packet []byte) error {
 func (t *Wrapper) Unwrap() tun.Device {
 	return t.tdev
 }
+
+var (
+	metricPacketIn              = clientmetric.NewGauge("tstun_in")
```

Comment:
"in" and "out" are not very descriptive. Just like "inbound" and "outbound", I will always have to look up what frame of reference the original author was using. Can you use absolute locators instead, e.g. {to,from}_{kernel,wg} or somesuch?

Reply:
I dunno, to me the frame of reference is `tailscaled`, the entity that's doing the logging and counting. `iptables` calls it `INPUT` and `OUTPUT` and that's not confusing? Is the confusion here that there's also wireguard involved?

Also, `in` and `out` match what we call `filterIn` and `filterOut`. And all the tstun hooks are named with `In$` and `Out$` names.

I can't think of something that's unambiguous for you but also not distractingly long.

I guess maybe `tstun_in_from_wg` as the prefix, so then `tstun_in_from_wg_drop_self_disco`?

And `tstun_out_to_wg` and `tstun_out_to_wg_drop_filter`?

You like those?

Reply:
The thing is, tailscaled has two inputs and outputs: one with the kernel, and one with the tailnet. And in different places in our code, we call either one in/out :/

So when I read "in", I have to figure out: did this mean "in from the tailnet", or "in from the kernel" ? Likewise when I see "out". This is inside tailscaled, so I can't just say "well it's into tailscaled", I also have to figure out into tailscaled _from where_. iptables doesn't have that problem as much, because all its inputs and outputs are the same kind, so it can reason about them as a bundle. But for us, kernel vs. tailnet is an important distinction, because kernel in == talinet out, usually.

I already get confused by tstun's in/out, and empirically I'm not alone given that we initially put the disco looping bugfix on the wrong leg of tstun :P 

I don't really know how to resolve this without making unwieldy names :/ Maybe we can shorten them further like: `tstun_wg_in` and `tstun_tun_in` ? That's unambiguously providing both a point of origin and a direction, so I don't have to go read the code to know where the counter is sitting?

Reply:
> I already get confused by tstun's in/out, and empirically I'm not alone given that we initially put the disco looping bugfix on the wrong leg of tstun :P

Well, we had little clue what macOS was doing and were largely guessing.

> `tstun_wg_in` and `tstun_tun_in`

Those names confuse me more. "wg_in" can mean "incoming from wireguard from the tailnet" or "we're feeding it in to wireguard to send out". Likewise with "tun".

My compromise with `tstun_in_from_wg` and `tstun_out_to_wg` was that they still started with `in` and `out` like before.

You can pick the suffixes after `tstun_in_` and `tstun_out_`, but let's stick with "in" and "out" prefixes and just agree on what in/out mean in a tstun context where those terms are already widely used. The metrics are largely intended to debug which paths we're flowing through the code, so it makes sense to name them after the code.

And in the context of this code, "in" means another a packet is arriving from somewhere else (in from the tailnet) and "out" means we're sending a packet like iptables OUTPUT (out to the tailnet, or out to 100.100.100.100).